src/anim/controller/anim-blend-tree.js

```diff
@@ -204,6 +94,176 @@ class AnimBlendTree extends AnimNode {
         }
         return count;
     }
+
+    static between(num, a, b, inclusive) {
+        var min = Math.min(a, b),
+            max = Math.max(a, b);
+        return inclusive ? num >= min && num <= max : num > min && num < max;
+    }
+
+    static getAngleRad(a, b) {
+        return Math.atan2(a.x * b.y - a.y * b.x, a.x * b.x + a.y * b.y);
+    }
+
+    static clamp(num, min, max) {
+        return num <= min ? min : num >= max ? max : num;
+    }
+
+}
+
+/**
+ * @private
+ * @class
+ * @name AnimBlendTree1D
+ * @classdesc An AnimBlendTree that calculates it's weights using a 1D algorithm
+ * @description Create a new BlendTree1D.
+ */
+class AnimBlendTree1D extends AnimBlendTree {
+    calculateWeights() {
+        var i;
+        var parameterValue = this._findParameter(this._parameters[0]).value;
+        if (this.parametersEqual([parameterValue])) return;
+        this._parameterValues = [parameterValue];
+        this._children[0].weight = 0.0;
+        for (i = 0; i < this._children.length - 1; i++) {
+            var c1 = this._children[i];
+            var c2 = this._children[i + 1];
+            if (AnimBlendTree.between(parameterValue, c1.point, c2.point, true)) {
```

Comment:
this feels like children should be stored in sorted order in the array .. just double checking that's not missing